### DISCUSSION TITLE
Custom domain verification? No TXT record shown

### DISCUSSION THREAD

**Discussion by ackledotdev** [community/community#59077] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi,

I'm trying to configure a custom subdomain for my GitHub Pages site. I am following steps from the [docs](https://docs.github.com/pages/configuring-a-custom-domain-for-your-github-pages-site/). When I save the domain (did the CNAME myself), GitHub shows a red banner at the top saying
> You must verify your domain discog.localplayer.dev before being able to use it. Check out https://docs.github.com/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages for more information.

I read the docs for domain verification, but the screenshots and the rest of the docs seem to be out of date. I don't see anything where I can add a TXT record ([step 5](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages#:~:text=wish%20to%20verify%20and%20select%20Add%20domain.-,Follow%20the%20instructions%20under%20%22Add%20a%20DNS%20TXT%20record%22%20to%20create%20the%20TXT%20record%20with%20your%20domain%20hosting%20service.,-Wait%20for%20your%20DNS%20configuration%20to%20change%2C%20this)).

**Comment by airtower-luna** [comment]
You need to add the TXT record to your domain, not anywhere in your repository. Exactly how to do that depends on your registrar, and how they let you configure DNS records.

**Reply by ackledotdev** [reply]
Yes, but what is the TXT record to add?
I can't see that anywhere.

**Reply by airtower-luna** [reply]
Ah, that wasn't obvious from the phrasing. The documentation doesn't mention any way to get it other than through the form, so I'd try to delete the `CNAME` file and add the domain from the settings instead of adding the file directly.

Or report a bug to support: https://support.github.com/contact/bug-report

**Reply by ackledotdev** [reply]
Thanks, I'll try that.

**Comment by dhrubasaha08** [comment]
@akpi816218 I think the documentation you are referring to might be outdated. GitHub Pages has updated its domain verification process, and adding a TXT record is no longer required. Here's the updated process to verify your custom subdomain for GitHub Pages:
- Create a CNAME file: In your repository, create a file named `CNAME` (no extension) in the root directory. The `CNAME` file should contain your custom subdomain, such as `discog.localplayer.dev`, on a single line.
- Configure DNS settings: Access your domain registrar or DNS provider's settings for the `localplayer.dev` domain and configure the DNS records for your custom subdomain. You will need to add a CNAME record pointing to your GitHub Pages domain. The CNAME record should have the following configuration:
  - Hostname/Name: Enter your desired subdomain (e.g., `discog`).
  - Value/Points to/Target: Enter your GitHub Pages domain (e.g., `yourusername.github.io`).
  - Save the DNS settings.
- Wait for DNS propagation: DNS changes may take some time to propagate across the internet. It can range from a few minutes to several hours. You need to wait for the changes to propagate before moving on to the next step.
- Verify domain in GitHub: Once DNS propagation is complete, go to your repository's settings page.
- Scroll down to the "GitHub Pages" section. In the "Custom domain" field, enter your custom subdomain (`discog.localplayer.dev`).
- Click on "Save".

At this point, GitHub Pages will attempt to verify the custom subdomain automatically. If everything is set up correctly and DNS propagation is complete, the verification process should succeed, and your custom subdomain will be associated with your GitHub Pages site.

Please note that it may take some time for the changes to take effect due to DNS propagation. If the verification doesn't succeed immediately, I recommend waiting for some time and trying again later. As an example, you can visit my [GitHub page](https://github.com/dhrubasaha08/dhrubasaha08.github.io) where I have successfully implemented a custom domain.

**Reply by ackledotdev** [reply]
Yes, this makes much more sense in context with the GitHub Pages repo settings GUI!
I'll try this, thanks so much.

**Reply by haha-eu** [reply]
@dhrubasaha08 Hi there, as stated here _"The `CNAME` file should contain your custom subdomain, such as `discog.localplayer.dev`, on a single line."_, where would I find my subdomain? Is that the same of my .github.io domain or is that something completley different? 

By the way, this is coming from a newbie to this whole world of coding and GitHub, so any help is apprectiaed.

**Reply by ackledotdev** [reply]
> @dhrubasaha08 Hi there, as stated here _"The `CNAME` file should contain your custom subdomain, such as `discog.localplayer.dev`, on a single line."_, where would I find my subdomain? Is that the same of my .github.io domain or is that something completley different? 
> 
> By the way, this is coming from a newbie to this whole world of coding and GitHub, so any help is apprectiaed.

Well your page is by default served on your `<username>.github.io` subdomain. If you want to access it from `discog.localplayer.dev`, that's your custom subdomain. If you want it served on `foo.bar.dev`, that's what you put on your CNAME.

**Reply by haha-eu** [reply]
@akpi816218 So do I just copy paste discog.localplayer.dev, and foo.bar.dev into the DNS settings, or is there something that I need to do with the code?

**Reply by ackledotdev** [reply]
> @akpi816218 So do I just copy paste discog.localplayer.dev, and foo.bar.dev into the DNS settings...

Yes, but replace with your own domain.
Are you hosting on GitHub Pages?

**Reply by haha-eu** [reply]
@akpi816218 I am sorry, but I am very confused. I am hosting on GitHub Pages, and I am using a custom domain. If the domain that I wanted to connect was https://comedy.eu.org, what would I need to do?\

**Reply by ackledotdev** [reply]
No problem with confusion :)
I assume your page is hosted on https://haha-eu.github.io/repo-name, right?

**Reply by haha-eu** [reply]
Yes. Do you want the actual name?

**Reply by haha-eu** [reply]
It is published here: https://haha-eu.github.io/comedy-eu/

**Reply by ackledotdev** [reply]
@haha-eu 
So then you would create two records:
- A file named `CNAME` exactly (no typos, otherwise it won't work) containing exactly `comedy.eu.org` (no https://)
- Another `CNAME` record with your DNS provider

I see that eu.org is a project dedicated to providing free subdomains. They probably have people who know to add such a record. I'm not familiar with the registration process they use, so I can't help you with that part. If you reach out to support and tell them you need a `CNAME` record at `comedy.eu.org` pointing to `haha-eu.github.io` (do not include the repo name, just the subdomain)

Wait until DNS propagation is mostly complete (use https://whatsmydns.net, as long as most of the DNS servers have the record it's ok)
Then you should go to [your user settings for GitHub Pages](https://github.com/settings/pages) and add a domain. It should give you a subdomain and a TXT record. Have them create a second subdomain, which should be something like `_github-pages-challenge-haha-eu.comedy`, with a `TXT` record with the value specified by GitHub (which would look like some hash of lowercase letters and numbers)
Wait for checks to complete (reload the GitHub settings page)

Once verification is complete, go to [the GitHub Pages settings for your repository](https://github.com/haha-eu/comedy-eu/settings/pages) and add `comedy.eu.org` in the Custom subdomain box. Click Save. You should see something saying "Wait for DNS checks to complete." After a while, that should change to "DNS check successful" in green with a check mark.

Navigate to https://comedy.eu.org to make sure the process worked.

If you have any more questions, please don't hesitate to reply.

**Reply by haha-eu** [reply]
Oh my god, thanks so much for this! I will let you know if I have any other questions.

**Reply by haha-eu** [reply]
@akpi816218, what would I put in the `CNAME` record in my DNS provider?

**Reply by ackledotdev** [reply]
`haha-eu.github.io`
make sure not to include "https://" nor "/comedy-eu/", just the subdomain

**Reply by linrongbin16** [reply]
hi, @dhrubasaha08 ,

In pages verifying, it shows "Add a DNS TXT record", but how to **add a DNS TXT record** ?

I didn't see any buttons, or links.

**Reply by airtower-luna** [reply]
That depends on how you operate DNS for your domain. If you're not sure how, chances are you need to check with the service where you registered the domain.

**Reply by linrongbin16** [reply]
hi @airtower-luna,

does it mean that, "DNS TXT record" is something that need to pay? for example I need to buy a domain service on AWS/Azure or some cloud service to make it work?

**Reply by airtower-luna** [reply]
You need to at least pay for the *domain* itself. Many registrars can handle the nameserver for you, and you just need to set the records you want (TXT and otherwise). Make sure to learn how DNS works before buying a domain or you won't get much use out of it. :wink:

**Reply by DrewPlayzYT** [reply]
> * Configure DNS settings: Access your domain registrar or DNS provider's settings for the `localplayer.dev` domain and configure the DNS records for your custom subdomain. You will need to add a CNAME record pointing to your GitHub Pages domain. The CNAME record should have the following configuration:
>   
>   * Hostname/Name: Enter your desired subdomain (e.g., `discog`).
>   * Value/Points to/Target: Enter your GitHub Pages domain (e.g., `yourusername.github.io`).
>   * Save the DNS settings.
what is a domain registrar?

**Comment by ackledotdev** [comment, marked as answer]
See [this answer](https://github.com/orgs/community/discussions/59077#discussioncomment-6277209), with one slight correction:
> Verify domain in GitHub: Once DNS propagation is complete, go to your repository's settings page.
DO NOT search for domain verification in *repo* settings. Go to your [user GitHub Pages settings](https://github.com/settings/pages), where you can verify your domain.

**Comment by mhekel** [comment]
I have a similar question: i have 2 domains on the same server (nwnmenzo.pw and nwnmenzo.com) the have their own dns reciords..when I add the txt record for nwnmenzo.com, Github successfully verifies it. I go back to github and enter nwnmenzo.pw and get the txt info, enter the txt record on my server, but github can't verify it. I'm about to conclude it doesn't like the .pw extension... i've had a couple other site reject it as well... any suggestions
note i did it for my personal account _github-pages-challenge-mhekel.nwnmenzo.pw. and later for my org's account _github-pages-challenge-SamsServerMenzo.nwnmenzo.pw. both to no avail. but the .com on both worked

**Reply by denikus** [reply]
same problem with .dev domain. Did you manage to solve it?

**Comment by raymelon** [comment]
Is this valuable information not in GitHub Docs? Really curious because it should have been included in "Verifying your custom domain for GitHub Pages" section, if that makes sense:

https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages
